### PR TITLE
feat: jdbc cross build for Scala 3

### PR DESCRIPTION
**Pull request by sebastian-alfers**

### CODE REVIEW

File: akka-projection-core-test/src/test/scala/akka/projection/internal/metrics/tools/InternalProjectionStateMetricsSpec.scala [hunk 1]
```diff
@@ -134,25 +134,32 @@ object InternalProjectionStateMetricsSpec {
     val adaptedHandlerStrategy: HandlerStrategy = offsetStrategy match {
       case ExactlyOnce(_) =>
         handlerStrategy match {
-          case SingleHandlerStrategy(handlerFactory) => {
+          case singleHandlerStrategy: SingleHandlerStrategy[Envelope] @unchecked => {
```
Comment (sebastian-alfers) [inline akka-projection-core-test/src/test/scala/akka/projection/internal/metrics/tools/InternalProjectionStateMetricsSpec.scala:137]: Scala3 complained here, so this was the only way I could make it compile, same below.

File: akka-projection-core-test/src/test/scala/akka/projection/internal/metrics/tools/InternalProjectionStateMetricsSpec.scala [hunk 2]
```diff
@@ -134,25 +134,32 @@ object InternalProjectionStateMetricsSpec {
...
+          case groupedHandlerStrategy: GroupedHandlerStrategy[Envelope] @unchecked => {
             val adaptedHandler = () =>
               new Handler[immutable.Seq[Envelope]] {
                 override def process(envelopes: immutable.Seq[Envelope]): Future[Done] =
-                  handlerFactory().process(envelopes).flatMap { _ =>
-                    offsetStore.saveOffset(projectionId, envelopes.last.offset)
-                  }
+                  groupedHandlerStrategy
+                    .handlerFactory()
+                    .asInstanceOf[Handler[Seq[Envelope]]]
```
Comment (patriknw) [inline akka-projection-core-test/src/test/scala/akka/projection/internal/metrics/tools/InternalProjectionStateMetricsSpec.scala:153]: shouldn't that be `immutable.Seq`? needed for Scala 2.12

Comment (patriknw) [reply akka-projection-core-test/src/test/scala/akka/projection/internal/metrics/tools/InternalProjectionStateMetricsSpec.scala:153]: btw, isn't it strange that this cast is needed? The `groupedHandlerStrategy` already has the `Envelope` type

**Review by patriknw — COMMENTED** [review]
looking good

File: akka-projection-core-test/src/test/scala/akka/projection/internal/metrics/tools/InternalProjectionStateMetricsSpec.scala (hunk 2 above)
Comment (sebastian-alfers) [reply akka-projection-core-test/src/test/scala/akka/projection/internal/metrics/tools/InternalProjectionStateMetricsSpec.scala]: Dang, good catch. This was a leftover with my battle with `@unchecked` which I did not revert.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**